docker/build.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+# Install build deps
+apk -U --no-progress add linux-pam-dev go@community gcc musl-dev
+
+# Init go environment to build Gogs
+mkdir -p ${GOPATH}/src/github.com/gogits/
+ln -s /app/gogs/ ${GOPATH}/src/github.com/gogits/gogs
+cd ${GOPATH}/src/github.com/gogits/gogs
+go get -v -tags "sqlite redis memcache cert pam"
+go build -tags "sqlite redis memcache cert pam"
+
+# Cleanup GOPATH
+rm -r $GOPATH
+
+# Remove build deps
+apk --no-progress del linux-pam-dev go gcc musl-dev
```

Comment:
## Gogs Building script

This script externalize the building process of gogs from the `Dockerfile`.

The purpose of this is to reduce the size of the container by having the build dependencies only installed in the container during building. We could just create a full block inside the `Dockerfile` but having a specific script seems way cleaner.

This little trick helps us save more that 200mo on the final container.